strawberry/asgi/__init__.py

```diff
@@ -54,9 +55,9 @@ async def get_root_value(self, request: Request) -> typing.Optional[typing.Any]:
         return None
 
     async def get_context(
-        self, request: typing.Union[Request, WebSocket]
+        self, request: typing.Union[Request, WebSocket], response: typing.Optional[Response] = None
```

Comment:
Will the `response` ever be `None`? It looks like in `handle_http` a response object is always being passed to this function.

Reply:
Without setting that default value to `None`, there were some failing tests...

```
tests/asgi/test_subscription.py 😰

================================================================================================================== FAILURES ==================================================================================================================
__________________________________________________________________________________________________________ test_simple_subscription __________________________________________________________________________________________________________

test_client = <starlette.testclient.TestClient object at 0x10863ba90>

    def test_simple_subscription(test_client):
        asyncio.set_event_loop_policy(TickEventLoopPolicy())

        with test_client.websocket_connect("/", "graphql-ws") as ws:
            ws.send_json({"type": GQL_CONNECTION_INIT})
            ws.send_json(
                {
                    "type": GQL_START,
                    "id": "demo",
                    "payload": {"query": "subscription { example }"},
                }
            )

            response = ws.receive_json()
            assert response["type"] == GQL_CONNECTION_ACK

>           response = ws.receive_json()

tests/asgi/test_subscription.py:37:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../Library/Caches/pypoetry/virtualenvs/strawberry-graphql-XTqJkshp-py3.9/lib/python3.9/site-packages/starlette/testclient.py:356: in receive_json
    message = self.receive()
../../Library/Caches/pypoetry/virtualenvs/strawberry-graphql-XTqJkshp-py3.9/lib/python3.9/site-packages/starlette/testclient.py:341: in receive
    raise message
../../Library/Caches/pypoetry/virtualenvs/strawberry-graphql-XTqJkshp-py3.9/lib/python3.9/site-packages/starlette/testclient.py:300: in _run
    loop.run_until_complete(self.app(scope, receive, send))
/usr/local/Cellar/python@3.9/3.9.2_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/asyncio/base_events.py:642: in run_until_complete
    return future.result()
strawberry/asgi/__init__.py:50: in __call__
    await self.handle_websocket(scope=scope, receive=receive, send=send)
strawberry/asgi/__init__.py:101: in handle_websocket
    async_result = await self.start_subscription(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <tests.asgi.conftest.GraphQL object at 0x10863bd30>, data = {'query': 'subscription { example }'}, operation_id = 'demo', websocket = <starlette.websockets.WebSocket object at 0x10863be20>

    async def start_subscription(self, data, operation_id: str, websocket: WebSocket):
        query = data["query"]
        variables = data.get("variables")
        operation_name = data.get("operation_name")

        if self.debug:
            pretty_print_graphql_operation(operation_name, query, variables)

>       context = await self.get_context(websocket)
E       TypeError: get_context() missing 1 required positional argument: 'response'

strawberry/asgi/__init__.py:136: TypeError
========================================================================================================== short test summary info ===========================================================================================================
FAILED 😰  tests/asgi/test_subscription.py::test_simple_subscription - TypeError: get_context() missing 1 required positional argument: 'response'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
======================================================================================================== 1 failed, 39 passed in 1.67s ========================================================================================================
```

Reply:
Oh I see, it doesn't play nice with websockets. Shame but ok.